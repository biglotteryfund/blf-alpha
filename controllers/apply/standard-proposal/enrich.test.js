/* eslint-env jest */
'use strict';
const { mockResponse } = require('./mocks');
const { enrichPending, enrichSubmitted } = require('./enrich');

test('enrich pending applications', function() {
    const result = enrichPending({
        id: 'some-uuid',
        formId: 'standard-enquiry',
        createdAt: '2020-03-04T23:00:00.000Z',
        expiresAt: '2020-03-04T23:00:00.000Z',
        updatedAt: '2020-03-04T23:00:00.000Z',
        applicationData: mockResponse()
    });

    expect(result).toMatchSnapshot({
        progress: expect.any(Object)
    });
});

test('enrich submitted applications', function() {
    const result = enrichSubmitted({
        id: 'some-uuid',
        formId: 'awards-for-all',
        createdAt: '2020-03-04T23:00:00.000Z',
        salesforceSubmission: {
            application: mockResponse()
        }
    });

    expect(result).toMatchSnapshot();
});
