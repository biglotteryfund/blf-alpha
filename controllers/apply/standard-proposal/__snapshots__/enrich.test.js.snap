// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enrich pending applications 1`] = `
Object {
  "amountRequested": "£250,000",
  "createdAt": "2020-03-04T23:00:00.000Z",
  "deleteUrl": "/apply/your-funding-proposal/delete/some-uuid",
  "editUrl": "/apply/your-funding-proposal/edit/some-uuid",
  "expiresAt": "2020-03-04T23:00:00.000Z",
  "formId": "standard-enquiry",
  "id": "some-uuid",
  "isExpired": false,
  "overview": Array [
    Object {
      "label": "Project length",
      "value": "1 years",
    },
    Object {
      "label": "Location",
      "value": "Derbyshire",
    },
    Object {
      "label": "Organisation",
      "value": "Example trading name",
    },
  ],
  "progress": Any<Object>,
  "projectCountry": "england",
  "projectName": "My project",
  "type": "pending",
  "untitledName": "Untitled proposal",
  "updatedAt": "2020-03-04T23:00:00.000Z",
}
`;

exports[`enrich submitted applications 1`] = `
Object {
  "amountRequested": "£250,000",
  "formId": "awards-for-all",
  "id": "some-uuid",
  "overview": Array [
    Object {
      "label": "Project length",
      "value": "1 years",
    },
    Object {
      "label": "Location",
      "value": "Derbyshire",
    },
    Object {
      "label": "Organisation",
      "value": "Example trading name",
    },
  ],
  "projectCountry": "england",
  "projectName": "My project",
  "submittedAt": "2020-03-04T23:00:00.000Z",
  "type": "submitted",
  "untitledName": "Untitled proposal",
}
`;
